__main__.py

```diff
@@ -88,11 +79,31 @@ def on_message_sent(future):
             future.result()
 
             print()
```

Comment:
With this refactoring, we're now spreading out the printing over multiple places in the code. In particular, you're printing out the message text in the `on_event_handler`, then printing out a newline here. That's a lot of tight coupling across otherwise separate functions.

In a future commit, we should change the error log to use the `logging` module, so it won't be "printing" per se. And here, we should just remove the printing of the newline and if that's really necessary, move it to where you print the message text.